activiti-cloud-service-common/activiti-cloud-service-messaging-config/src/main/java/org/activiti/cloud/common/messaging/config/FunctionBindingConfiguration.java

```diff
@@ -188,4 +198,24 @@ public Object postProcessAfterInitialization(Object bean, String beanName) throw
     }
 
     public interface BindingResolver extends Function<String, String> {}
+
+    @Bean
+    @ConditionalOnClass(JacksonMapper.class)
+    @Primary
+    public JacksonMapper jacksonMapper(@Autowired(required = false) ObjectMapper objectMapper) {
+        //temporary workaround for https://github.com/spring-cloud/spring-cloud-function/issues/1159
+        ObjectMapper copiedMapper;
+        try {
+            copiedMapper = objectMapper.copy();
```

Comment:
As `objectMapper` is optional, it's better check if it's `null` before calling `copy` method to avoid potential NPE. If it's null you can already instantiate a new one.

Reply:
NPE will be handled in the same way as an issue with copying existing objectMapper that's why I did not checked it for null

Reply:
Yes, I know that the case of NPE is already covered by the generic catch exception, but usually using exceptions as control flow is considered as an anti-pattern and the check for null is easy to put in place, so I'd prefer avoiding NPE, rather than throwing + catching the NPE.